### PR TITLE
Table Sorting Enhancements

### LINKED ISSUE #219: Sorting indicator support for the Table component when binding with custom data
**Is your feature request related to a problem? Please describe.**
Not related to an issue; request for enhancement

**Describe the solution you'd like**
Would be great to have the table show sort indicator (column/direction) when binding with the custom data.  Would also like to have an event raised from the component when user changes the column for sorting or changes the sort direction (from ascending to descending and vice versa).  Component should also accept the sort column and direction for the initial render.  After the initial render the sort column and direction would be managed from the header row within the Table component.  Only some of the columns would be sortable and at a time only one column would be sortable.  This is for situations where the filtering and sorting of the data is done at the micro-services layer to make sure UI only binds to the data that it needs and no more.

**Describe alternatives you've considered**
No alternatives considered.  If there are viable alternatives to achieve similar functionality that adheres to USWDS guidelines please recommend.

**Additional context**
none

### PR DESCRIPTION
**Pull request by jbouder**
Provide additional functionality to the USWDS Table.

## Description

- Added optional sortable property to TableColumn definition
- Updated table to apply sorting to columns based on new sortable property, and default to true if not specified
- Added onSort prop to Table to support running some function when sort changes
- Fixed unrelated unique key issue in step indicator

## Related Issue

Resolves #219 

## Motivation and Context

- Enhance sort-related configurability of the UWDS Table

## How Has This Been Tested?
- Local testing

## Screenshots (if appropriate):
Screenshot 2024-08-05 at 11 00 45 AM

### CODE REVIEW

File: packages/comet-uswds/src/components/step-indicator/step-indicator.tsx [hunk 1]
```diff
@@ -76,7 +76,7 @@ export const StepIndicator = ({
           }
 
           return (
-            <li {...attributes} key={`usa-step-label-${id}`}>
+            <li {...attributes} key={`usa-step-label-${stepIndex}`}>
```
Comment (jbouder) [inline packages/comet-uswds/src/components/step-indicator/step-indicator.tsx:79]: This is unrelated to this PR, fixing an issue from a recent update.

File: packages/comet-uswds/src/components/table/table.tsx [hunk 2]
```diff
@@ -114,18 +140,25 @@ export const Table = ({
...
-                aria-sort={sortable && sortIndex === index ? sortDir : undefined}
-              >
-                {column.name}
-              </th>
-            ))}
+            {columns
+              .map((obj) => ({
+                ...obj,
+                sortable: obj.sortable !== undefined ? obj.sortable : true,
+              }))
```
Comment (jbouder) [inline packages/comet-uswds/src/components/table/table.tsx:147]: If sortable not provided, default to true

**Review by kildre — APPROVED** [review]
Approved
